File: src/main/java/com/pubnub/api/models/consumer/files/PNListFilesResult.java
package com.pubnub.api.models.consumer.files;

import com.pubnub.api.models.consumer.PNPage;
import lombok.Data;
import lombok.NonNull;

import java.util.Collection;

@Data
public class PNListFilesResult {
    @NonNull
    private final int count;
    private final PNPage.Next next;
    @NonNull
    private final int status;
    @NonNull
    private final Collection<PNUploadedFile> data;
}

File: core/src/main/kotlin/com/pubnub/api/models/consumer/files/PNListFilesResult.kt
package com.pubnub.api.models.consumer.files

import com.pubnub.api.models.consumer.objects.PNPage

data class PNListFilesResult(
    val count: Int,
    val next: PNPage.PNNext?,
    val status: Int,
    val data: Collection<PNUploadedFile>
)
